drivers/timer/nrf_grtc_timer.c

```diff
@@ -396,16 +358,22 @@ int z_nrf_grtc_wakeup_prepare(uint64_t wake_time_us)
 	nrfx_err_t err_code;
 	static uint8_t systemoff_channel;
 	uint64_t now = counter();
+	nrfx_grtc_sleep_config_t sleep_cfg;
 	/* Minimum time that ensures valid execution of system-off procedure. */
-	uint32_t minimum_latency_us = nrfy_grtc_waketime_get(NRF_GRTC) +
-				      nrfy_grtc_timeout_get(NRF_GRTC) +
-				      CONFIG_NRF_GRTC_SLEEP_MINIMUM_LATENCY;
+	uint32_t minimum_latency_us;
 	uint32_t chan;
 	int ret;
 
+	nrfx_grtc_sleep_configuration_get(&sleep_cfg);
+	minimum_latency_us = (sleep_cfg.waketime + sleep_cfg.timeout) * USEC_PER_SEC / 32768 +
```

Comment:
where 32768 comes from? Shouldn't it be a define? Can it depend on some prescaler?

Reply:
> where 32768 comes from? Shouldn't it be a define? Can it depend on some prescaler?

it comes from LFXO which is always 32768. Also WAKETIME and TIMEOUT register's unit is 32ki cycles

Reply:
The code should not be obscure. Instead, you could use `#define LFCLK_FREQUENCY_HZ 32768` or `#define LFCLK_TICKS_PER_SEC 32768` or `GRTC_RTCOUNTER_TICKS_PER_SEC`. Then it would be clear why you divide by this number.

Like the constant `USEC_PRE_SEC` you use in this line.